src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1373 @@
+                    // This might require an integer register to store the value.
+                    buildInternalIntRegisterDefForNode(blkNode);
+                }
+
+                const bool isDstRegAddrAlignmentKnown = dstAddr->OperIsLocalAddr();
+
+                if (isDstRegAddrAlignmentKnown && (size > FP_REGSIZE_BYTES))
+                {
+                    // TODO-LoongArch64: For larger block sizes CodeGen can choose to use 16-byte SIMD instructions.
+                    // here just used a temp register.
+                    buildInternalFloatRegisterDefForNode(blkNode);
+                }
+            }
+            break;
+
+            case GenTreeBlk::BlkOpKindHelper:
+                assert(!src->isContained());
+                dstAddrRegMask = RBM_ARG_0;
+                srcRegMask     = RBM_ARG_1;
+                sizeRegMask    = RBM_ARG_2;
+                break;
+
+            default:
+                unreached();
+        }
+    }
+    else
+    {
+        if (src->OperIs(GT_IND))
+        {
+            assert(src->isContained());
+            srcAddrOrFill = src->AsIndir()->Addr();
+        }
+
+        if (blkNode->OperIs(GT_STORE_OBJ))
+        {
+            // We don't need to materialize the struct size but we still need
+            // a temporary register to perform the sequence of loads and stores.
+            // We can't use the special Write Barrier registers, so exclude them from the mask
+            regMaskTP internalIntCandidates =
+                allRegs(TYP_INT) & ~(RBM_WRITE_BARRIER_DST_BYREF | RBM_WRITE_BARRIER_SRC_BYREF);
+            buildInternalIntRegisterDefForNode(blkNode, internalIntCandidates);
+
+            if (size >= 2 * REGSIZE_BYTES)
+            {
+                // TODO-LoongArch64: We will use ld/st paired to reduce code size and improve performance
+                // so we need to reserve an extra internal register.
+                buildInternalIntRegisterDefForNode(blkNode, internalIntCandidates);
+            }
+
+            // If we have a dest address we want it in RBM_WRITE_BARRIER_DST_BYREF.
+            dstAddrRegMask = RBM_WRITE_BARRIER_DST_BYREF;
+
+            // If we have a source address we want it in REG_WRITE_BARRIER_SRC_BYREF.
+            // Otherwise, if it is a local, codegen will put its address in REG_WRITE_BARRIER_SRC_BYREF,
+            // which is killed by a StoreObj (and thus needn't be reserved).
+            if (srcAddrOrFill != nullptr)
+            {
+                assert(!srcAddrOrFill->isContained());
+                srcRegMask = RBM_WRITE_BARRIER_SRC_BYREF;
+            }
+        }
+        else
+        {
+            switch (blkNode->gtBlkOpKind)
+            {
+                case GenTreeBlk::BlkOpKindUnroll:
+                {
+                    buildInternalIntRegisterDefForNode(blkNode);
+
+                    const bool isSrcAddrLocal = src->OperIs(GT_LCL_VAR, GT_LCL_FLD) ||
```

Comment:
I thought this was `NYI` for now?

Reply:
Thanks very much.
I will update it within the next patch.

Reply:
> I thought this was `NYI` for now?

I have tested on LA,
this `buildInternalIntRegisterDefForNode` should be left.

The related `isSrcAddrLocal` can be deleted.

Thanks.